### ISSUE TITLE
Serialization of custom OT types

### ISSUE THREAD

**Issue by DetweilerRyan** [share/ShareJS#389]
Hi ShareJS community!

You guys have done an amazing job here and I'd like to contribute.

I am exploring the possibility of using ShareJS to solve the shared-mutable-state problem I have with my Trello-like web app.  I have concluded that the JSON0 type cannot be used to enforce the invariants of my application state and therefore I must implement a custom OT type for my application state.  

The snapshot for my custom OT type is an immutable object from the Immutable.js npm module.  As a result, I need custom serialization/deserialization methods as documented here: https://github.com/ottypes/docs#optional-properties.

Unfortunately, the shareJS and liveDB libraries do not support the optional serialization/deserialization methods on OT types.  I have forked the ShareJS and LiveDB repos to add custom serialization for my purposes ( https://github.com/DetweilerRyan/ShareJS https://github.com/DetweilerRyan/livedb ).

I feel there is a strong use case for custom serialization support ( I can go into more detail if needed ) and believe it should be supported by the two libraries.

**Comment by josephg** [comment]
Yep, I totally agree. I've been wanting to add that for awhile. Make a pull request and I'll pull it in.

**Comment by DetweilerRyan** [comment]
Thanks for the response.  Once I feel comfirtable with my implementation, I'll make the pull request.
